### PR TITLE
[CBRD-23293] Fix conversion function for numeric type

### PR DESCRIPTION
**Pull request by lucianlazar2211**
http://jira.cubrid.org/browse/CBRD-23293

Added a new function for the conversion of numeric type.

_EDIT: new function is equivalent of ldr_int_elem_

### CODE REVIEW

File: src/loaddb/load_db_value_converter.cpp [hunk 1]
```diff
@@ -946,4 +947,67 @@ namespace cubload
...
     return ER_FAILED;
   }
+
+  int
+  to_int_generic (const char *str, const attribute *attr, db_value *val)
+  {
+    size_t str_len = strlen (str);
+    int error_code = NO_ERROR;
+
+    db_make_int (val, 0);
```
Comment (eseokoh) [inline src/loaddb/load_db_value_converter.cpp]: please move it inside of `if`.

File: src/loaddb/load_db_value_converter.cpp [hunk 2]
```diff
@@ -946,4 +947,67 @@ namespace cubload
...
+
+    /* Let try take the fastest path here, if we know that number we are getting fits into a long, use strtol, else we
+     * need to convert it to a double and coerce it, checking for overflow. Note if integers with leading zeros are
```
Comment (eseokoh) [inline src/loaddb/load_db_value_converter.cpp]: please correct the comment. it looks like to_db_int's.

File: src/loaddb/load_db_value_converter.cpp (hunk 1 above)
Comment (ribram) [inline src/loaddb/load_db_value_converter.cpp]: Move `db_make_int` to `if (str_len < MAX_DIGITS_FOR_INT || ...)` branch

File: src/loaddb/load_db_value_converter.cpp [hunk 3]
```diff
@@ -946,4 +947,67 @@ namespace cubload
...
+	DB_NUMERIC num;
+	DB_BIGINT tmp_bigint;
+
+	numeric_coerce_dec_str_to_num (str, num.d.buf);
+	if (numeric_coerce_num_to_bigint (num.d.buf, 0, &tmp_bigint) != NO_ERROR)
+	  {
+	    error_code = db_value_domain_init (val, DB_TYPE_NUMERIC, str_len, 0);
+	    if (error_code != NO_ERROR)
+	      {
+		er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IT_DATA_OVERFLOW, 1, pr_type_name (DB_TYPE_BIGINT));
```
Comment (ribram) [inline src/loaddb/load_db_value_converter.cpp]: why is this an overflow error?

File: src/loaddb/load_db_value_converter.cpp [hunk 4]
```diff
@@ -946,4 +947,67 @@ namespace cubload
...
+		return ER_IT_DATA_OVERFLOW;
+	      }
+
+	    error_code = db_value_put (val, DB_TYPE_C_CHAR, (char *) str, str_len);
+	    if (error_code != NO_ERROR)
+	      {
+		er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IT_DATA_OVERFLOW, 1, pr_type_name (DB_TYPE_BIGINT));
```
Comment (ribram) [inline src/loaddb/load_db_value_converter.cpp]: Same question here.

File: src/loaddb/load_db_value_converter.cpp [hunk 5]
```diff
@@ -946,4 +947,67 @@ namespace cubload
...
+		return ER_IT_DATA_OVERFLOW;
+	      }
+
+	    return error_code;
```
Comment (ribram) [inline src/loaddb/load_db_value_converter.cpp]: unnecessary return; let it fall to `return NO_ERROR`

File: src/loaddb/load_db_value_converter.cpp (hunk 3 above)
Comment (eseokoh) [reply src/loaddb/load_db_value_converter.cpp]: the only case that the function returns an error is str_len > max precision of numeric.
I'm not sure this holds.

File: src/loaddb/load_db_value_converter.cpp [hunk 6]
```diff
@@ -946,4 +947,68 @@ namespace cubload
...
+	    error_code = db_value_put (val, DB_TYPE_C_CHAR, (char *) str, str_len);
+	    if (error_code != NO_ERROR)
+	      {
+		ASSERT_ERROR ();
+		return error_code;
+	      }
+	  }
+	else
+	  {
+	    val->data.bigint = tmp_bigint;
```
Comment (ribram) [inline src/loaddb/load_db_value_converter.cpp]:
```suggestion
	    db_make_bigint (val, tmp_bigint);
```

**Pull request merged**